blockstack/lib/config.py

```diff
@@ -284,8 +289,128 @@ def get_genesis_block_stages():
 def get_all_genesis_block_patches():
     return GENESIS_BLOCK_PATCHES
 
+def load_genesis_bulk_address_lines(path, line_start, line_end):
+    """
+    Read a list of lines from a file, from the given starting line (inclusive) to the given ending line (exclusive).
+    The file should be a sequence of newline-separated b58check addresses.
+    Aborts on failure.
+    """
+    if not os.path.exists(path):
+        print >> sys.stderr, 'FATAL: no such file or directory: "{}"'.format(path)
+        os.abort()
+
+    f = open(path, 'r')
+    line_ptr = 0
+    ret = []
+    while True:
+        line = f.readline()
+        if len(line) == 0:
+            # EOF
+            break
+
+        if line_ptr >= line_start and line_ptr < line_end:
+            line = line.strip()
+            addr = b58ToC32(line)
+            ret.append(addr)
+        
+        line_ptr += 1
+        if line_ptr >= line_end:
+            break
+
+    if line_ptr != line_end:
+        # underrun
+        print >> sys.stderr, 'FATAL: underrun reading from "{}": EOF at line {}, expected {}'.format(path, line_ptr, line_end)
+        os.abort()
+
+    f.close()
+    return ret
+
+def check_genesis_bulk_patch_integrity(path, expected_sha256):
+    h = hashlib.new('sha256')
+    f = open(path, 'r')
+    while True:
+        buf = f.read(65536)
+        if len(buf) == 0:
+            break
+
+        h.update(buf)
+
+    return h.hexdigest() == expected_sha256
+
+
+def get_genesis_bulk_address_path(path):
+    """
+    Get the absolute path to a data file, given its path's name
+    """
+    import blockstack.data
+    dirname = os.path.dirname(blockstack.data.__file__)
+    return os.path.join(dirname, path)
+    
+
+def load_genesis_bulk_patch(file_patch):
+    from schemas import GENESIS_BLOCK_ROW_SCHEMA
+
+    # format: {'path': '/path/to/patch', 'sha256': ..., 'line_start': start_line, 'line_end': end_line, 'receive_whitelisted': ..., 'lock_send': ..., 'metadata': ..., 'type': ..., 'value': ..., 'vesting': ..., 'vesting_total': ...}
```

Comment:
[nit] might be easier to read as a multi-line comment with indented json

Reply:
Fixed